src/Package/Impl/ProjectSystem/PropertyPages/PropertyPageViewModel.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem;
+
+namespace Microsoft.VisualStudio.R.Package.ProjectSystem.PropertyPages {
+    internal abstract class PropertyPageViewModel : INotifyPropertyChanged {
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        /// <summary>
+        /// Special text value that indicates that the string property values for the 
+        /// selected configurations are not identical.
+        /// </summary>
+        internal static readonly string DifferentStringOptions = Resources.PropertyPageDifferentOptions;
+
+        /// <summary>
+        /// Special value that indicates that the bool property values for the selected
+        /// configurations are not identical.
+        /// </summary>
+        internal static readonly bool? DifferentBoolOptions = null;
+
+        public UnconfiguredProject UnconfiguredProject { get; set; }
+
+        public PropertyPageControl ParentControl { get; set; }
+
+        /// <summary>
+        /// The project properties for the currently selected project configurations.
+        /// </summary>
+        public IRProjectProperties[] ConfiguredProjectsProperties { get; set; }
+
+        /// <summary>
+        /// Since calls to ignore events can be nested, a downstream call could change the outer 
+        /// value.  To guard against this, IgnoreEvents returns true if the count is > 0 and there is no setter. 
+        /// PushIgnoreEvents\PopIgnoreEvents  are used instead to control the count.
+        /// </summary>
+        private int _ignoreEventsNestingCount = 0;
+
+        public bool IgnoreEvents { get { return _ignoreEventsNestingCount > 0; } }
+
+        public void PushIgnoreEvents() {
+            _ignoreEventsNestingCount++;
+        }
+
+        public void PopIgnoreEvents() {
+            Debug.Assert(_ignoreEventsNestingCount > 0);
+            if (_ignoreEventsNestingCount > 0) {
+                _ignoreEventsNestingCount--;
+            }
+        }
+
+        public abstract Task Initialize();
+
+        public abstract Task<int> Save();
```

Comment:
`Save` is used by `PropertyPageControl`, which doesn't know about the derived class.